buildroot/share/PlatformIO/variants/BIGTREE_GENERIC_STM32F407_5X/variant.h

```diff
@@ -222,9 +222,15 @@ extern "C" {
 //#define DACC_RESOLUTION         12
 
 // PWM resolution
-#define PWM_RESOLUTION          8
-#define PWM_FREQUENCY           20000
-#define PWM_MAX_DUTY_CYCLE      255
+/*
+ * BEWARE:
+ * Changing this value from the default (1000) will affect the PWM output value of analogWrite (to a PWM pin)
+ * Since the pin is toggled on capture, if you change the frequency of the timer you have to adapt the compare value (analogWrite thinks you did)
+ */
+//#define PWM_FREQUENCY           20000
```

Comment:
If there are situations requiring the PWM base frequency to be changed (i.e., high-frequency diode lasers), then if possible we should make the compare value adapt accordingly without requiring manually changing it.

Reply:
@thinkyhead that define is actually used only inside the STM32 HAL for analogwrite.
We could modify it back to 20,000 but if we do so we should hal a marlin wrapper for analogwrite which recalculates the value for the new frequency.
Anyway, if in some circumstances a higher frequency for the pwm is needed there's a hal method to set it `analogWriteFrequency`.

Reply:
Makes sense. In fact, if the defines are no longer used or needed for the HAL there is no need to keep them around. We can just delete all these lines.

Reply:
wait. don't. the actual STM32 template has those lines (with the defaults, instead of 20.000) I would leave them there to inform the devs that those values can be changed if needed (as STM did). the only difference from the default is that 20,000 which I left as a comment because I didn't know if there was a reason why someone changed it to 20,000 so "ok, this is causing the problem, I comment the line and add the reason why I commented it"
(if the one who changed it to 20,000 did the same maybe we could now do a brainstorm if reverting it back to the default is good or not...but we don't know why it has been done. there were no comments, it was creating problems in 5.7.0 and I reverted it back.)

Reply:
> to inform the devs that those values can be changed if needed

But if they are not actually used, then changing them makes no difference. It looks to me that the whole concept around these defines is being killed off.

Reply:
Sorry, it was not clear that these defines pertain to the platform and not to the Marlin HAL.